nodejs/eks/addons/addon.ts

```diff
@@ -37,17 +37,16 @@ export class Addon extends pulumi.ComponentResource {
     constructor(name: string, args: AddonOptions, opts?: pulumi.CustomResourceOptions) {
         const cluster = args.cluster;
 
-        super("eks:index:Addon", name, args, {
-            ...opts,
+        super("eks:index:Addon", name, args, pulumi.mergeOptions(opts, {
             parent: cluster,
-        });
+        }));
 
         const addon = new aws.eks.Addon(
             name,
             {
                 ...args,
                 clusterName: cluster.core.cluster.name,
-                configurationValues: JSON.stringify(args.configurationValues),
+                configurationValues: stringifyAddonConfiguration(args.configurationValues),
```

Comment:
We might need to be more careful here. `stringifyAddonConfiguration` changes the ordering, and may cause replacements for existing users. We'll need to test this out.